Authentication pull request

sergiocortessat opened the pull request:
The request corresponds to the Authentication activity which includes devise

Pull request for the Forms project.

We have added the required parts of the project. As such it includes Sign_in and Sign_up sessions. The Users have the ability to make many Posts. Therefore having many to many relations on the User and Post model. The Project complies with the requirements of creating posts with automatic generate user_id keys. We have decided to set the page to only display the name of the Author who is signed in. But can't see any of the other author's name. This is a personal adding as the project requires only a user to be signed_in to see all author's name.

Logic has been moved away from the views using partials and helpers.

This page will only display an index page with all posts. the ability to sign in  Sign_up and create new posts

Commit:
Add devise initial set_up

nganifaith committed:
Add the post model and controller.

Commit:
Add new.html and new partial form

Commit:
Add _forms, helpers for logic.

Commit:
Add helpers logic

nganifaith committed:
Add logic using helpers

nganifaith committed:
Add bootstrap file

nganifaith committed:
Add custom css to post file

nganifaith committed:
Add header partial

Commit:
Add style, Readme

Commit:
Add  lino to edit account

Commit:
updates for heroku deployment

Commit:
Add live version to readme, linters fix

nganifaith committed:
Change ruby version in the Gemfile

nganifaith committed:
Fix stylelint errors

mke2111 reviewed (changes requested):
# STATUS: CHANGES REQUIRED 🔴 ♻️ 

## Hey Happy Coders, 👋🏽 
**This is a great project you have going on, well-done lads**. However, there are a few issues to fix to get approved.

### REQUIRED CHANGES: ⚠️ 

- [x] When I use the URL listed in the readme file, no repository is found. Replace the URL you have currently in the clone section of the readme file with the correct https url.
- [x] It is better to create new posts using the `build` method instead of the `create` method. Change to `@post = current_user.posts.build(post_params)` in the create action of post controllers.
- [x] The application layout should have Sign in / Sign out links (So they are not manually repeated in every view) [Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_in_out_link_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49722686/)


- [x] [OPTIONAL] There is a typo in the readme section for the Prerequisites. Give this a quick fix.
- [x] [OPTIONAL] Your app should display Flash messages for unsuccessful actions.


### Kindly make all the requested changes and submit another CRR. 💯 

**Cheers 🍹 and Happy coding 💻** 

## Ping me on slack for a chit-chat 🆘  Roy Mukuye Or mention me in this PR

Commit:
Add fixes _header, .build command

Commit:
Add flash_notices and typo readme

sergiocortessat commented:
Pull request 2. 

We have added all the required and optional changes to the repo.

_header moved to layouts. 
Flash notices for controller actions.
 .build method instead of .create for post_controller created method
Fixed typo errors

Thank you

nganifaith committed:
Update flash messages in post controller

oloomoses reviewed (approved):
# Congratulations, Project Approved. :muscle: :tada: :clap: 
**Great job working on the suggested changes from the previous review.
You have passed all the requirements for this project :muscle: 
Feel free to merge and close this pull request.**

**I wish you the best in your next project and see you at the top :rocket:** 
#### Happy Coding
_Slack: Oloo Moses_
